src/dma/traits.rs

```diff
@@ -19,64 +19,76 @@ pub trait SafePeripheralRead {}
 
 /// Trait for DMA stream interrupt handling.
 pub trait StreamISR: crate::Sealed {
-    /// Clear all interrupts for the DMA stream.
-    fn clear_interrupts(&mut self);
+    /// Clear all interrupts flags for the DMA stream.
+    fn clear_all_flags(&mut self);
 
     /// Clear transfer complete interrupt (tcif) for the DMA stream.
-    fn clear_transfer_complete_interrupt(&mut self);
+    fn clear_transfer_complete_flag(&mut self);
 
-    /// Clear half transfer interrupt (htif) for the DMA stream.
-    fn clear_half_transfer_interrupt(&mut self);
+    /// Clear half transfer interrupt flag (htif) for the DMA stream.
+    fn clear_half_transfer_flag(&mut self);
 
-    /// Clear transfer error interrupt (teif) for the DMA stream.
-    fn clear_transfer_error_interrupt(&mut self);
+    /// Clear transfer error interrupt flag (teif) for the DMA stream.
+    fn clear_transfer_error_flag(&mut self);
 
-    /// Clear direct mode error interrupt (dmeif) for the DMA stream.
-    fn clear_direct_mode_error_interrupt(&mut self);
+    /// Clear direct mode error interrupt flag (dmeif) for the DMA stream.
+    fn clear_direct_mode_error_flag(&mut self);
 
-    /// Clear fifo error interrupt (feif) for the DMA stream.
-    fn clear_fifo_error_interrupt(&mut self);
+    /// Clear fifo error interrupt flag (feif) for the DMA stream.
+    fn clear_fifo_error_flag(&mut self);
+
+    /// Get all interrupts flags a once.
+    ///
+    /// The tuple contain in order:
+    ///  - transfer complete flag
+    ///  - half transfer flag
+    ///  - transfer error flag
+    ///  - direct mode error flag
+    ///  - fifo_error flag
+    fn all_flags(&self) -> DmaFlags;
 
     /// Get transfer complete flag.
-    fn get_transfer_complete_flag() -> bool;
```

Comment:
I understand the reason of deleting prefix, but it is hard to read code without it.
Possibly it is better to use `is_` prefix here for consistency with PAC's bit reading names.

Reply:
maybe i can also remove the "_flag" suffix ?
```Rust
fn is_transfer_complete(&self) -> bool;
fn clear_transfer_complete(&mut self) ->bool;
```

Reply:
Yes. Summarizing:
`is_smth`
`clear_smth`
`listen_smth`
`unlisten_smth`

Reply:
cc @therealprof

Reply:
done for all interrupt individually